### PR TITLE
Add reset-password and hash-password commands

### LINKED ISSUE #2499: New "sensuctl user" subcommands & field
### Description
https://github.com/sensu/sensu-go/pull/3753 introduced two new subcommands and a new field in the `User` struct.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
- https://docs.sensu.io/sensu-go/latest/api/users/
- https://docs.sensu.io/sensu-go/latest/reference/rbac/#user-specification
- https://docs.sensu.io/sensu-go/latest/reference/rbac/#manage-users
- https://docs.sensu.io/sensu-go/latest/sensuctl/reference/#change-admin-user-s-password (?)

### Expected behavior
There 2 new subcommands:

1. `sensuctl user hash-password`, which generates a password hash for the given cleartext password.
2. `sensuctl user reset-password`, which resets a user password without having to specify the current password; this requires explicit permission to the `users` resource, so typically only cluster-admins will be able to do that.

Additionally, it adds the `password_hash` field to the `User` struct. The `sensuctl user hash-password` command can be used to determine a hash for a given password. With that, you can ignore the `password` field and only specify the hash, and then `sensuctl create/dump/edit` users without having any knowledge of the passwords.

### Additional context

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds `reset-password` and `hash-password` commands in sensuctl reference
- Adds `User` and `user` as sensuctl create types
- Adds links to `reset-password` and `hash-password` commands in user section of RBAC reference

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2499

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@palourde I'm having trouble sorting out how the three password commands work.

1. When I use hash-password for a user I created, I get:
```
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-enterprise-go % sensuctl user hash-password ali
$5X$10$XXXXXXgzTpUsgdfyEOf/cOMSOvDjd9UWEGIoy.VNVRnG./XXXXXXX
```
But when I GET /users/ali, I do not see a `password_hash` field:
```
{
    "username": "ali",
    "groups": [
        "ops"
    ],
    "disabled": false
}
```

Where should I be seeing the `password_hash` field? Or am I using the sensuctl command incorrectly?

2. When I run `sensuctl user reset-password ali --interactive`, I get the interactive prompts for new password and confirm password and confirmation that it is updated:
```
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-enterprise-go % sensuctl user reset-password ali --interactive
? New Password:		 ********
? Confirm Password:	 ********
Updated
```
Yet when I enter this password for `sensuctl user change-password ali --interactive`:
```
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-enterprise-go % sensuctl user change-password ali --interactive
? Current Password:	 ********
Error: could not find field matching current-password
```

@palourde one more bit of information: I think the reset-password command worked because I was able to log in to the web UI as user `ali` with the password I used. So perhaps the problem is the `change-password` command is somehow not picking up the new password I set with `reset-password`?

**Comment by palourde** [comment]
@hillaryfraley 

1. `sensuctl user hash-password [PASSWORD]` generates a hash for the given password locally; it does not modify or even communicate with the password. In your example, you generated a hash for the password `ali`.

That being said, you won't see the `password_hash` field via the API at the moment, I was mistaken; If you dump or just list users, you won't see their hashed passwords. There's some security concerns with that, so it would probably require some discussions before doing that.

2. It appears to be a bug with the interactive mode of this command, I just created https://github.com/sensu/sensu-go/pull/3836. You should however be able to do the same with the flags.

File: content/sensu-go/5.21/sensuctl/reference.md [hunk 1]
```diff
@@ -83,6 +83,29 @@ Run:
...
+
+#### Generate a password hash
+
+To generate a password hash for the given cleartext password, run:
+
+{{< highlight shell >}}
+sensuctl user hash-password
+{{< /highlight >}}
+
+The `sensuctl user hash-password` command adds a field to the `User` resource, `password_hash`, whose value is a hash for the user's password.
```
Comment (echlebek) [inline content/sensu-go/5.21/sensuctl/reference.md]: The `sensuctl user hash-password` command creates a bcrypt hash of a given password.

(It does not add a field to the User resource, although one was added for this release)

File: content/sensu-go/5.21/sensuctl/reference.md [hunk 2]
```diff
@@ -1276,3 +1300,6 @@ Flags are optional and apply only to the `delete` command.
 [48]: #sensuctl-prune-resource-types
 [49]: #examples
 [50]: ../../reference/webconfig/
+[51]: #update-resources
+[52]: #export-resources
+[53]: ../../reference/rbac/#user-specification
```
Comment (hillaryfraley) [inline content/sensu-go/5.21/sensuctl/reference.md:1305]:
```suggestion
[53]: ../../reference/rbac/#user-specification
[54]: https://en.wikipedia.org/wiki/Bcrypt
```

File: content/sensu-go/5.21/sensuctl/reference.md (hunk 1 above)
Comment (hillaryfraley) [inline content/sensu-go/5.21/sensuctl/reference.md]:
```suggestion
The `sensuctl user hash-password` command creates a [bcrypt hash][54] of the password.
```

File: content/sensu-go/5.21/sensuctl/reference.md [hunk 3]
```diff
@@ -83,6 +83,29 @@ Run:
...
+
+#### Generate a password hash
+
+To generate a password hash for the given cleartext password, run:
+
+{{< highlight shell >}}
+sensuctl user hash-password
+{{< /highlight >}}
+
+The `sensuctl user hash-password` command creates a [bcrypt hash][54] of the password.
```
Comment (hillaryfraley) [inline content/sensu-go/5.21/sensuctl/reference.md]:
```suggestion
The `sensuctl user hash-password` command creates a [bcrypt hash][54] of the specified password.
```

File: content/sensu-go/5.21/sensuctl/reference.md [hunk 4]
```diff
@@ -83,6 +83,29 @@ Run:
...
+{{< /highlight >}}
+
+You must have admin permissions to use the `sensuctl user reset-password` command.
+
+#### Generate a password hash
+
+To generate a password hash for the given cleartext password, run:
+
+{{< highlight shell >}}
+sensuctl user hash-password
```
Comment (hillaryfraley) [inline content/sensu-go/5.21/sensuctl/reference.md]:
```suggestion
sensuctl user hash-password PASSWORD
```

File: content/sensu-go/5.21/sensuctl/reference.md [hunk 5]
```diff
@@ -83,6 +83,29 @@ Run:
...
+
+{{< highlight shell >}}
+sensuctl user reset-password USERNAME --interactive
+{{< /highlight >}}
+
+You must have admin permissions to use the `sensuctl user reset-password` command.
+
+#### Generate a password hash
+
+To generate a password hash for the given cleartext password, run:
```
Comment (hillaryfraley) [inline content/sensu-go/5.21/sensuctl/reference.md]:
```suggestion
To generate a password hash for a specified cleartext password, run:
```

**Review by palourde — APPROVED** [review]
👍

**Review by echlebek — APPROVED** [review]

**Pull request merged**